x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/cache/blob/BlobStoreCacheMaintenanceService.java

```diff
@@ -97,22 +226,13 @@ private ShardRouting systemIndexPrimaryShard(final ClusterState state) {
         return null;
     }
 
-    private static boolean hasSearchableSnapshotWith(final ClusterState state, final SnapshotId snapshotId, final IndexId indexId) {
+    private static boolean hasSearchableSnapshotWith(final ClusterState state, final String snapshotId, final String indexId) {
         for (IndexMetadata indexMetadata : state.metadata()) {
             final Settings indexSettings = indexMetadata.getSettings();
             if (SearchableSnapshotsSettings.isSearchableSnapshotStore(indexSettings)) {
-                final SnapshotId otherSnapshotId = new SnapshotId(
-                    SNAPSHOT_SNAPSHOT_NAME_SETTING.get(indexSettings),
-                    SNAPSHOT_SNAPSHOT_ID_SETTING.get(indexSettings)
-                );
-                if (Objects.equals(snapshotId, otherSnapshotId)) {
-                    final IndexId otherIndexId = new IndexId(
-                        SNAPSHOT_INDEX_NAME_SETTING.get(indexSettings),
-                        SNAPSHOT_INDEX_ID_SETTING.get(indexSettings)
-                    );
-                    if (Objects.equals(indexId, otherIndexId)) {
-                        return true;
-                    }
+                if (Objects.equals(snapshotId, SNAPSHOT_SNAPSHOT_ID_SETTING.get(indexSettings))
```

Comment:
Note: this method does not check if the snapshot belongs to the same repository. Since snapshot UUID and index UUID are unique ids I think the risk of collision is very limited, given that in case of collision the cached docs are not deleted.